### PR TITLE
Refactor Shell Scripts based on ShellCheck's suggestion

### PR DESCRIPTION
**Pull request by PeterDaveHello**
This will help improve the robustness of the scripts, some of the deprecated and non-safe statements need to be refactored.

### CODE REVIEW

**Comment by Changaco** [comment]
Thanks but I'm not going to merge this as-is. The changes seem mostly unnecessary, and some of them are in obsolete files that I should have deleted months ago.

**Comment by PeterDaveHello** [comment]
I can revert the changes on files need to be deleted. I don't think they are all that unnecessary, for example, one of their header used `#!/bin/sh` which may not support many of statements in the script, and `-o`/`-a` in bash condition were deprecated for a while, it's no hurt to have these changes, they only make those scripts less potential of issues.

**Comment by whit537** [comment]
Merging the changes to the obsolete files makes no difference since the files are obsolete anyway.

Merging the unnecessary changes probably(?) won't hurt.

I agree that @PeterDaveHello's PRs such as this one tend to be low-value, but ¯\\\_(ツ)\_/¯. Either merge or close to keep the PR queue moving.

**Comment by Changaco** [comment]
Rebased on master and added a commit to delete the obsolete files.

**Comment by Changaco** [comment]
And another to revert changes that weren't merely unnecessary but invalid.

File: release.sh [hunk 1]
```diff
@@ -1,13 +1,13 @@
-#!/bin/sh
+#!/bin/bash
 
 # Fail on errors and undefined variables
 set -eu
 
 # Be somewhere predictable
-cd "`dirname $0`"
+cd "$(dirname "$0")"
```
Comment (whit537) [inline release.sh]: I don't know how the double quotes behave here.

**Comment by whit537** [comment]
Assuming those double double quotes lines work as intended, lgtm.

File: release.sh (hunk 1 above)
Comment (Changaco) [reply release.sh]: This is quite common in shell scripts, I think it works as intended.

**Pull request merged**